File: programme/random-odd-number/random-odd-number.py
# Write a programme to generate a random odd number in range n1-n2

import random


def random_odd_number(number1, number2):
    random_number = random.randint(number1, number2)

    while random_number % 2 == 0:
        random_number = random.randint(number1, number2)
    return random_number


range_start = 0
range_end = 100
print("Start range :", range_start)
print("End   range :", range_end)
print("\nRandom odd number :", random_odd_number(range_start, range_end))

File: programme/random-odd-number/random-odd-number.c
// Write a programme to generate a random odd number in range n1-n2

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
  int lower, upper, num;

  srand(time(0));
  printf("Enter lower limit : ");
  scanf("%d", &lower);
  printf("Enter upper limit : ");
  scanf("%d", &upper);

  int check = 2 % 2;

  printf("\nThe random numbers is : ");

  while (check == 0)
  {
    num = (rand() % (upper - lower + 1)) + lower;
    check = num % 2;
  }
  printf("%d\n", num);

  return 0;
}
